pkg/application/downloader.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package application
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+)
+
+type Downloader interface {
+	Download(url string) ([]byte, error)
+	GetLatestReleaseVersion(releaseURL string) (string, error)
+}
+
+type downloader struct{}
+
+func NewDownloader() Downloader {
+	return &downloader{}
+}
+
+func (downloader) Download(url string) ([]byte, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected http status code: %d", resp.StatusCode)
+	}
+
+	return io.ReadAll(resp.Body)
+}
+
+// GetLatestReleaseVersion returns the latest available version from github
+func (downloader) GetLatestReleaseVersion(releaseURL string) (string, error) {
```

Comment:
A suggestion just for you to consider and decide: I think we are already using different downloaders, maybe we could have this be just a real downloader, without any "business" logic, downloading from the supplied URL, and then do version checks and stuff in a wrapper?

Reply:
Oh I see, this is basically just moved around. So we don't have maybe that many Downloaders after all.